Firestore/Example/Tests/API/FSTAPIHelpers.mm

```diff
@@ -154,7 +155,8 @@
                             mutatedKeys,
                             static_cast<bool>(fromCache),
                             /*sync_state_changed=*/true,
-                            /*excludes_metadata_changes=*/false};
+                            /*excludes_metadata_changes=*/false,
+                            hasCachedResults ? true : false};
```

Comment:
if hasCachedResults type is boolean, why we need to check it here? Are inputs like `BOOL hasCachedResults` nullable and thats why we add `hasCachedResults ? true : false` to check existence?

Reply:
I thought the same thing. It turns out that Objective-C's `BOOL` type cannot be converted to C++'s `bool` type. That being said, I see a few lines above (line 156) they use `static_cast` to convert `fromCache` from `BOOL` to `bool`; therefore, I'll change the code to do _that_.